### PR TITLE
Version 0.1.9

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
While trying to build something with V, a number of things were frustrating me about the developer experience. This PR fixes some of them.

I suggest testing it out manually on your own computer since this is probably more changes than normal. I also have not tested this on Windows.

Here is the changelog I wrote:

>
>   - Improve `v fmt` support. This extension now registers as the default formatter for V in VSCode.
>   - V Language Server now starts up lazily rather than eagerly. Before, it started up in any workspace which contained `.v` files. Now it only starts up if one of the tabs is a V tab and a file persisted to disk
>   - Fixed a bug where sometimes `v fmt` would run before saving to disk. This changes it to copy the document to stdin instead, so there's no chance of that happening.
>   - `v fmt` now logs syntax errors in the output and shows a checkmark. If the V Language Server is not running and formatting failed due to a syntax error, the output window will appear with the error message (easier to read than a popup)
>   - Multiple workspace folders in the same window should be a little more reliable
>   - Remove features that just didn't work
>   - Non-userfacing: switched from webpack to esbuild to make this extension build faster
>

As a bonus:
- I copy-pasted the changelog into the readme, so that the changelog is visible from inside the VSCode Extension Store
- Added a little documentation on how to use VLS with this
- Removed mention of features that were not implemented or were removed at some point (such as tests)
- Switched the extension JS bundler from webpack to esbuild because its faster
- Re-added the devbits playground

Some of this diff is just Prettier automatically formatting. Sorry about that.

### CODE REVIEW

**Comment by danieldaeschle** [comment]
Thanks for this.

Could you please provide each of these changes in single PRs instead of one big? It's easier to review and easier to pick what we want.

1. We won't use vfmt as formatter. We will use vls to format files. So this change won't be accepted and was removed with a reason.
2. Please don't do version bumps. That's the job of the team/maintainer.
3. esbuild seems ok, but i don't see any benefit. The build was fast enough so far. But it's ok.
4. Why did you readd the devbits playground? That's not really the goal of this extension. We try to keep the extension as small as possible.

**Comment by serkonda7** [comment]
Further the changelog is already visible in VS Code in the `Changes` tab of the extension so it should be removed from the readme.

**Comment by Jarred-Sumner** [comment]
> We won't use vfmt as formatter. We will use vls to format files. So this change won't be accepted and was removed with a reason.

I respect that this is ultimately not my call, since this is not my project and I'm not the maintainer. As a user though, `vls` unfortunately doesn't seem ready yet to replace `v fmt`. It would be fantastic if it was (and I'm sure it will be someday!). However, auto-formatting in VSCode hasn't worked for ~months https://github.com/vlang/vscode-vlang/issues/212#issuecomment-788134824. After this change, it works (and comes with a status bar). Prioritizing the needs of the present versus preparing for the future is a tricky balance.

> Please don't do version bumps. That's the job of the team/maintainer.

My apologies -- I definitely overstepped here.

> esbuild seems ok, but i don't see any benefit. The build was fast enough so far. But it's ok.

Motivation was that to test the extension, I manually ran this command repeatedly:
```bash
npm run compile && npm run package && code --install-extension  vscode-vlang-0.1.9.vsix
```

It was faster with esbuild and took ~2m to switch. That being said, it would be great to have unit tests in the extension itself (especially running `vls` inside vscode as integration tests).

> Why did you readd the devbits playground? That's not really the goal of this extension. We try to keep the extension as small as possible.

I don't have a strong opinion on this. I saw mention in Discord that people recommended using devbits instead of the WASM one because the WASM version was very out of date.

> Further the changelog is already visible in VS Code in the Changes tab of the extension so it should be removed from the readme.

I never noticed that the Changes tab exists. Happy to ✂️ that

I don't have time to split this up into several small PRs. Feel free to close this if reviewing will be too much work or otherwise if the changes are overall not peaceful. I feel bad -- this was  immature of me to dump this giant PR on you guys

**Comment by JalonSolov** [comment]
Please don't feel bad.  Enthusiasm is a wonderful thing, to be encouraged rather than stifled.  The only problem is that there are only 2 active maintainers, and it is harder for them to deal with several things in a single PR, while actively doing development as well.

In general, for future reference, it is usually better to only do one change per PR, but you can do multiple PR that require they be merged in sequence.  And something you _have_ to do several changes at once because everything would "break" if done in pieces.

So TL;DR is please keep submitting changes, just try to break them up.  ;-)

**Comment by kahsa** [comment]
working vfmt _now_ is highly welcomed!

**Comment by danieldaeschle** [comment]
> Motivation was that to test the extension, I manually ran this command repeatedly:
Just run webpack-dev and press F5 and select VS Code Extension Development. There is no need to wait.

vfmt might be implmented too. But we have to find out how it behaves if two formatters are installed then... it's not that easy.

We appreciate if you split this PR up. Then most of the changes can be merged.
